src/libguac/client.c

```diff
@@ -132,7 +132,8 @@ guac_stream* guac_client_alloc_stream(guac_client* client) {
         return NULL;
 
     /* Allocate stream */
-    stream_index = guac_pool_next_int(client->__stream_pool);
+    stream_index = guac_pool_next_int_below_or_die(client->__stream_pool,
```

Comment:
Right above this we do a non-threadsafe check against `GUAC_CLIENT_MAX_STREAMS`, which will cause the function to return `NULL` - do we still need that? Does the new check not correctly handle this?

And if the documented behavior of this function is to return `NULL` when out of streams, why are we dying here instead?

It seems like we'd want to maintain that `NULL` return for backwards compatibility, but also I'd point out that none of the callers of these `alloc_stream()` functions look to actually check if the return value is non-NULL before using it...

Reply:
> Right above this we do a non-threadsafe check against GUAC_CLIENT_MAX_STREAMS, which will cause the function to return NULL - do we still need that? Does the new check not correctly handle this?

Yes - very good catch. I'll remove the superfluous check.

> And if the documented behavior of this function is to return NULL when out of streams, why are we dying here instead?

Also fair - will fix.

Reply:
Both should now be fixed.